queries/synths/useSynthBalances.ts

```diff
@@ -9,19 +9,18 @@ import { useRecoilState, useRecoilValue } from 'recoil';
 import QUERY_KEYS from 'constants/queryKeys';
 import Connector from 'containers/Connector';
 import { balancesState } from 'store/futures';
-import { networkState, walletAddressState } from 'store/wallet';
+import { networkState } from 'store/wallet';
 
 type SynthBalancesTuple = [string[], ethers.BigNumber[], ethers.BigNumber[]];
 
-const useSynthBalances = (options?: UseQueryOptions<Balances>) => {
-	const walletAddress = useRecoilValue(walletAddressState);
+const useSynthBalances = (address: string | null, options?: UseQueryOptions<Balances>) => {
```

Comment:
Yea we can revert this actually, I started using it for cross margin account balances but then realised it updates recoil state and is used in the header wallet component. This is the sort of thing it would be great to refactor by lifting shared state management to higher up in the app. It's also the issue behind this bug #1216